tests/unit/test_expression_tree/test_matrix.py

```diff
@@ -24,18 +24,13 @@ def test_matrix_evaluate(self):
 
     def test_matrix_operations(self):
         np.testing.assert_array_equal((self.mat + self.mat).evaluate(), 2 * self.A)
-        np.testing.assert_array_equal((self.mat - self.mat).evaluate(), 0 * self.A)
+        np.testing.assert_array_equal(
+            (self.mat - self.mat).evaluate().toarray(), 0 * self.A
+        )
         np.testing.assert_array_equal(
             (self.mat @ self.vect).evaluate(), np.array([[5], [2], [3]])
         )
 
-    def test_matrix_modification(self):
```

Comment:
Should this be removed? I don't see how it is related to simplifying on creation.

Reply:
simplifying on creation broke the test, then I thought we were never using this functionality so could get rid of the test. We might actually want to prevent matrix modification outright, to avoid weird bugs?